### DISCUSSION TITLE
[Suggestion] Option for daemonizing

### DISCUSSION THREAD

**Discussion by phantomcraft** [DNSCrypt/dnscrypt-proxy#1868] [Ideas]
I believe it would be very useful when running dnscrypt-proxy from shell scripts.

Here I have to use **daemonize** tool in my scripts when sending dnscrypt-proxy to background: https://github.com/bmc/daemonize

**Comment by jedisct1** [comment]
There used to be an option for it. However, daemonization is a complicated and uncommon thing in Go applications.

Existing libraries were found to be unreliable on non-Linux systems, so the feature was eventually removed.

`-service install` / `-service start` achieve the same thing, except that it persists on reboot and works on most platforms.
